make/test_target_script.sh

```diff
@@ -0,0 +1,248 @@
+    then
+        print_highlighted_msg $LIGHT_CYAN "The environment variable ERL_TOP has been set to the git root"
+    else
+        echo "The ERL_TOP environment variable need to be set before this script is executed."
+        exit 1
+    fi
+fi
+
+export ERL_TOP=$ERL_TOP
+
+
+if [ -z "${ARGS}" ]
+then
+   ARGS="$@"
+fi
+
+# make test in root
+DIR=`pwd`
+if [ "$DIR" -ef "$ERL_TOP" ]
+then
+    TARGET_SYS=`$ERL_TOP/erts/autoconf/config.guess`
+    REL_DIR="$ERL_TOP/release/$TARGET_SYS"
+    cd "$REL_DIR"
+    ./Install -sasl `pwd`
+    export PATH="$REL_DIR/bin:$PATH"
+    cd "$ERL_TOP/release/tests/test_server"
+    print_all_tests_takes_long_time_warning
+    echo "The tests will start in a few seconds..."
+    sleep 45
+    cd "$ERL_TOP/release/tests/test_server"
+    erl -eval "ts:install(),erlang:halt()"
+    erl -noinput -eval "ts:run([all_tests,batch]),erlang:halt()"
+    exit $?
+fi
+
+# make test in application directory
+if [ -d test ]
+then
+    APPLICATION=`basename $DIR`
+    cd test
+    echo "The tests in test directory for $APPLICATION will be executed with ct_run"
+    if [ -z "${ARGS}" ]
+    then
+        print_all_tests_for_application_notes
+        if find . -type f -name '*.c' | grep -q "."
+        then
+            print_c_files_warning
+        fi
+    fi
+    CT_RUN="$ERL_TOP/bin/ct_run"
+    MAKE_TEST_DIR="`pwd`/make_test_dir"
+    MAKE_TEST_REL_DIR="$MAKE_TEST_DIR/${APPLICATION}_test"
+    MAKE_TEST_CT_LOGS="$MAKE_TEST_DIR/ct_logs"
+    RELEASE_TEST_SPEC_LOG="$MAKE_TEST_CT_LOGS/release_tests_spec_log"
+    mkdir -p $MAKE_TEST_DIR
+    mkdir -p $MAKE_TEST_REL_DIR
+    mkdir -p $MAKE_TEST_CT_LOGS
+    make RELSYSDIR=$MAKE_TEST_REL_DIR release_tests_spec > $RELEASE_TEST_SPEC_LOG 2>&1
+    if [ $? != 0 ]
+    then
+        cat $RELEASE_TEST_SPEC_LOG
+        print_highlighted_msg $RED "\"make RELSYSDIR=$MAKE_TEST_REL_DIR release_tests_spec\" failed."
+        exit 1
+    fi
+    SPEC_FLAG=""
+    SPEC_FILE=""
+    if [ -z "${ARGS}" ]
+    then
+        SPEC_FLAG="-spec"
+        SPEC_FILE="$MAKE_TEST_REL_DIR/$APPLICATION.spec"
+        ARGS="$SPEC_FLAG $SPEC_FILE"
+    fi
+    # Compile test server
+    (cd "$ERL_TOP/lib/common_test/test_server" && make)
+    # Run ct_run
+    cd $MAKE_TEST_REL_DIR
+    $CT_RUN -logdir $MAKE_TEST_CT_LOGS\
+           -pa "$ERL_TOP/lib/common_test/test_server"\
+           ${ARGS}\
+           -erl_args\
+           -env "$PATH"\
+           -env ERL_CRASH_DUMP "$MAKE_TEST_DIR/${APPLICATION}_erl_crash.dump"\
+           -boot start_sasl\
+           -sasl errlog_type error\
+           -pz "$ERL_TOP/lib/common_test/test_server"\
+           -pz "."\
+           -ct_test_vars "{net_dir,\"\"}"\
+           -noshell\
+           -sname test_server\
+           -rsh ssh\
+           ${ERL_ARGS}
+    CT_RUN_STATUS=$?
+    if [ $CT_RUN_STATUS = "0" ]
+    then
+        print_highlighted_msg $GREEN "The test(s) ran successfully (ct_run returned a success code)"
+        exit 0
+    else
```

Comment:
Maybe print the error code? And I think it would be a good idea to print a link to the ct logs. i.e.

    echo file://$MAKE_TEST_CT_LOGS/index.html

Reply:
or at least exit with the same error code as ct_run.

Reply:
Both fixed in commit below.